src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitRouter.kt

```diff
@@ -133,7 +121,11 @@ abstract class AbstractRabbitRouter<T> : MessageRouter<T> {
      * @param message a source message which can be reduced according to pin configuration
      * @return the source message, part of them, or null if the message is matched to the pin configuration: fully, partially, not match accordingly
      */
-    protected abstract fun splitAndFilter(message: T, pinConfiguration: PinConfiguration): Collection<T>
+    protected abstract fun splitAndFilter(
+        message: T,
+        pinConfiguration: PinConfiguration,
+        pinName: PinName
+    ): Collection<T>
```

Comment:
This function can be return T because it reduces the source message by filter

Reply:
done